File: 646-maximum-length-of-pair-chain/646-maximum-length-of-pair-chain.py
class Solution:
    def findLongestChain(self, pairs: List[List[int]]) -> int:
        pairs = sorted(pairs, key=lambda x: x[1])
        curr, res = -1001, 0
        for a, b in pairs:
            if curr < a:
                curr = b
                res += 1
        return res

File: 646-maximum-length-of-pair-chain/646-maximum-length-of-pair-chain.js
/**
 * @param {number[][]} pairs
 * @return {number}
 */
var findLongestChain = function(pairs) {
  pairs.sort((a, b) => a[1] === b[1] ? a[0] - b[0] : a[1] - b[1]);
  let [curr, ans] = [-Infinity, 0];
  
  for (let [a, b] of pairs) {
    if (curr < a) {
      curr = b;
      ans++;
    }
  }
  
  return ans;
};